pkg/endpoint/endpoint.go

```diff
@@ -675,6 +676,19 @@ func (e *Endpoint) GetModel() *models.Endpoint {
 	return e.GetModelRLocked()
 }
 
+func (e *Endpoint) getDesiredL4Policy() (l4Policy *policy.L4Policy) {
+	if e.desiredPolicy != nil {
+		l4Policy = e.desiredPolicy.L4Policy
+	}
+	if l4Policy == nil && e.realizedPolicy != nil {
+		l4Policy = e.realizedPolicy.L4Policy
```

Comment:
Why would we want realizedPolicy to be desiredPolicy? This doesn't make sense to me.

Reply:
This was a remnant from my attempt to clear the desired L4Policy pointer when it has been realized. If no new desired policy was created, then it would have been the same as the realized policy. This approach did not work for other reasons.

Thanks for pointing this out, I removed this, as the e.desiredPolicy.L4Policy is now never cleared.